modules/composer-workflowoperation/src/main/java/org/opencastproject/workflow/handler/composer/CutMarksToSmilWorkflowOperationHandler.java

```diff
@@ -0,0 +1,319 @@
+    }
+
+    // Create the new SMIL document
+    Smil smil;
+    try {
+      SmilResponse smilResponse = smilService.createNewSmil(mediaPackage);
+
+      logger.info("Start Adding tracks");
+      for (Times mark : cutmarks) {
+        smilResponse = smilService.addParallel(smilResponse.getSmil());
+        SmilMediaContainer par = (SmilMediaContainer) smilResponse.getEntity();
+        logger.debug("Segment begin: {}; Segment duration: {}", mark.begin, mark.duration);
+        // Add tracks (as array) to par
+        smilResponse = smilService
+                .addClips(smilResponse.getSmil(),
+                        par.getId(),
+                        tracksFromFlavors.toArray(new Track[tracksFromFlavors.size()]), //new Track[] { presenterTrack, presentationTrack },
+                        mark.begin,
+                        mark.duration);
+      }
+
+      smil = smilResponse.getSmil();
+      logger.info("Done adding tracks");
+    } catch (SmilException e) {
+      throw new WorkflowOperationException("Failed to create SMIL Catalog", e);
+    }
+
+    // Put new SMIL into workspace and add it to mediapackage
+    try (InputStream is = IOUtils.toInputStream(smil.toXML(), "UTF-8")) {
+      URI smilURI = workspace.put(mediaPackage.getIdentifier().toString(), smil.getId(), CUTTING_SMIL_NAME, is);
+      MediaPackageElementBuilder mpeBuilder = MediaPackageElementBuilderFactory.newInstance().newElementBuilder();
+      Catalog catalog = (Catalog) mpeBuilder
+              .elementFromURI(smilURI, MediaPackageElement.Type.Catalog, targetSmilFlavor);
+      catalog.setIdentifier(smil.getId());
+      for (String tag : targetTags) {
+        catalog.addTag(tag);
+      }
+      mediaPackage.add(catalog);
+    } catch (JAXBException | SAXException | IOException e) {
+      throw new WorkflowOperationException("Failed to parse crated SMIL Catalog", e);
+    }
+
+
+    final WorkflowOperationResult result = createResult(mediaPackage, WorkflowOperationResult.Action.CONTINUE);
+    logger.debug("Cut marks to smil operation completed");
+    return result;
+  }
+
+  private Optional<Track> getTrackFromFlavor(MediaPackageElementFlavor flavor, MediaPackage mediaPackage)
+          throws WorkflowOperationException {
+    Optional<Track> result = Optional.empty();
+
+    // Get tracks from flavor
+    TrackSelector trackSelector = new TrackSelector();
+    trackSelector.addFlavor(flavor);
+    Collection<Track> tracks = trackSelector.select(mediaPackage, false);
+
+    // Get only videos
+    ArrayList<Track> videos = new ArrayList<Track>();
+    for (Track video : tracks) {
+      if (video.hasVideo()) {
+        videos.add((video));
+      }
+    }
+
+    // Check for number of videos to avoid any issues
+    // Possible TODO: Can we handle more than one track per flavor?
+    if (videos.isEmpty())
+      return result;
+    if (videos.size() > 1) {
+      for (Track track : videos) {
+        logger.info("Track {} in flavor {}", track, flavor);
```

Comment:
A flavor is no collection. It's an attribute:
```suggestion
        logger.info("Track {} with flavor {}", track, flavor);
```